lona/pytest.py

```diff
@@ -96,3 +99,56 @@ def setup_aiohttp_app(loop: AbstractEventLoop) -> Application:
         )
 
     return setup_lona_project_context
+
+
+def eventually(timeout: float = 5, interval: float = 1) -> Iterator[AsyncContextManager]:  # NOQA: LN001
+    """
+        Wait for expected state in async test.
+
+        The function is meant to be iterated. Each time it returns new attempt.
+        Attempt is an async context manager to wrap assertions.
+        It suppresses all exceptions until time is out.
+        If no exception is raised iteration stops.
+
+        Example::
+
+            counter = 0
+            for attempt in eventually():
+                async with attempt:
+                    counter += 1
+                    assert counter > 3
+
+        :param timeout: time in seconds during which it produces new attempts
+        :param interval: seconds to sleep between attempts
+    """
+
+    end_time = time.time() + timeout
+    success = False
+
+    while not success:
+        context_manager_was_used = False
+
+        @asynccontextmanager
+        async def attempt() -> AsyncIterator[None]:
+            nonlocal context_manager_was_used, success
+            context_manager_was_used = True
+
+            try:
+                yield  # execute assertions
+
+            except KeyboardInterrupt:
```

Comment:
Why are KeyboardInterrups excepted if they are reraised without further handling?

Reply:
I thought that it is necessary to be able interrupting tests. But It turned out that I was wrong. Removed this.